docs/prod-maintain-secure-upgrade.md

```diff
@@ -12,82 +12,103 @@ secure Zulip installation, including:
-  security patches.
-
-* To use the Zulip API with your Zulip server, you will need to use the
-  API endpoint of e.g. `https://zulip.example.com/api`.  Our Python
-  API example scripts support this via the
-  `--site=https://zulip.example.com` argument.  The API bindings
-  support it via putting `site=https://zulip.example.com` in your
-  .zuliprc.
-
-  Every Zulip integration supports this sort of argument (or e.g. a
-  `ZULIP_SITE` variable in a zuliprc file or the environment), but this
-  is not yet documented for some of the integrations (the included
-  integration documentation on `/integrations` will properly document
-  how to do this for most integrations).  Pull requests welcome to
-  document this for those integrations that don't discuss this!
-
-* Similarly, you will need to instruct your users to specify the URL
-  for your Zulip server when using the Zulip desktop and mobile apps.
-
-* As a measure to mitigate the impact of potential memory leaks in one
-  of the Zulip daemons, the service automatically restarts itself
-  every Sunday early morning.  See `/etc/cron.d/restart-zulip` for the
-  precise configuration.
+**We recommend reading this entire section before doing your first
+upgrade.**
+
+To upgrade to a new version of the zulip server, download the appropriate
+release tarball from
+[https://www.zulip.com/dist/releases/](https://www.zulip.com/dist/releases/)
+
+You also have the option of creating your own release tarballs from a copy of
+zulip.git repository using `tools/build-release-tarball`. And, starting with
+Zulip version 1.4, you can deploy updates [directly from a Git
+repository](#deploying-zulip-from-a-git-repository).
+
+Next, run as root:
+
+```
+/home/zulip/deployments/current/scripts/upgrade-zulip zulip-server-VERSION.tar.gz
+```
+
+The upgrade process will shut down the Zulip service and then run `apt-get upgrade`, a
+puppet apply, any database migrations, and then bring the Zulip service back
+up. Upgrading will result in some brief downtime for the service, which should be
+under 30 seconds unless there is an expensive transition involved. Unless you
+have tested the upgrade in advance, we recommend doing upgrades at off hours.
+
+### Preserving local changes to configuration files
+
+**Warning**: If you have modified configuration files installed by
+Zulip (e.g. the nginx configuration), the Zulip upgrade process will
+overwrite your configuration when it does the `puppet apply`.
+
+You can test whether this will happen assuming no upstream changes to
+the configuration using `scripts/zulip-puppet-apply` (without the
+`-f` option), which will do a test puppet run and output and changes
+it would make. Using this list, you can save a copy of any files
+that you've modified, do the upgrade, and then restore your
+configuration.
+
+If you need to do this, please report the issue so
+that we can make the Zulip puppet configuration flexible enough to
+handle your setup.
+
+### Upgrade log location
```

Comment:
I would consider using consistent verb tense for headings.  If headings apply to things users should do 90% of the time, then use imperative tense.  If headings apply to things users might inquire about, but which aren't necessary, then use the more passive gerund tense.  Maybe I'm confused here because "Upgrade" is a verb, but it's used an adjective.  So maybe say "Using the upgrade log for error reporting".